src/sentry/utils/snuba.py

```diff
@@ -96,6 +101,16 @@ def raw_query(start, end, groupby=None, conditions=None, filter_keys=None,
     if not project_ids:
         raise SnubaError("No project_id filter, or none could be inferred from other filters.")
 
+    # any project will do, as they should all be from the same organization
+    project = Project.objects.get(pk=project_ids[0])
+    retention = quotas.get_event_retention(
+        organization=Organization(project.organization_id)
+    )
+    if retention:
+        start = max(start, datetime.utcnow() - timedelta(days=retention))
+        if start >= end:
+            return {}
```

Comment:
Is it safe to return an empty dict? Do any callers expect certain keys to exist even if there are no results?